frontend/src/metabase/admin/settings/components/SettingsPages/GeneralSettingsPage.tsx

```diff
@@ -0,0 +1,75 @@
+import { jt, t } from "ttag";
+
+import { useDocsUrl } from "metabase/common/hooks";
+import ExternalLink from "metabase/core/components/ExternalLink";
+import { Stack } from "metabase/ui";
+
+import { AdminSettingInput } from "../widgets/AdminSettingInput";
+import { AnonymousTrackingInput } from "../widgets/AnonymousTrackingInput";
+import { CustomHomepageDashboardSetting } from "../widgets/CustomHomepageDashboardSetting";
+import { HttpsOnlyWidget } from "../widgets/HttpsOnlyWidget";
+import { SiteUrlWidget } from "../widgets/SiteUrlWidget";
+
+export function GeneralSettingsPage() {
+  const { url: iframeDocsUrl } = useDocsUrl("configuring-metabase/settings", {
+    anchor: "allowed-domains-for-iframes-in-dashboards",
+  });
+
+  return (
+    <Stack gap="xl" maw="42rem" px="lg" py="sm">
+      <AdminSettingInput
+        name="site-name"
+        title={t`Site Name`}
+        description={t`The name used for this instance of Metabase.`}
+        inputType="text"
+      />
+
+      <SiteUrlWidget />
```

Comment:
SiteUrlWidget returns null if `isHosted=true`. The same for `HttpsOnlyWidget`. 

For these conditional widgets, rather than defensively coding in each component maybe we should hoist up the conditions and short-circuit the rendering at the page level? 

HttpsWidget also checks isHttps, but I'm happy to keep that check in the component because it's so specific

Reply:
I really like having each component be responsible for its own logic, that way we can easily move them around and know we haven't dropped their display conditions.

Reply:
Okay, I can see what you mean!